File: frontend/src/models/Offer.ts
import { CourseDto } from "./Course";
import { OfferStatus } from "./enums/OfferStatus";

export interface OfferDto {
  offer_id: string;
  owner_username: string;
  receiver_username: string;
  state: OfferStatus;
  my_course: CourseDto;
  wanted_course: CourseDto;
}

File: backend/src/main/java/com/example/usos_oauth/offers/model/dao/Offer.java
package com.example.usos_oauth.offers.model.dao;

import com.example.usos_oauth.security.model.User;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "offer")
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "offer_seq")
    @SequenceGenerator(name = "offer_seq", allocationSize = 1)
    private long offerId;
    @ManyToOne
    private User owner;
    @ManyToOne
    private User receiver;
    @Enumerated(EnumType.STRING)
    private OfferState state = OfferState.PENDING;
    @ManyToOne(cascade = CascadeType.PERSIST)
    private Course myCourse;
    @ManyToOne(cascade = CascadeType.PERSIST)
    private Course wantedCourse;
    @OneToMany(cascade = CascadeType.ALL)
    private List<Comment> comments;
}
